libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -707,6 +708,23 @@ public void cancelFeedback(String feedbackId) {
     navigationTelemetry.cancelFeedback(feedbackId);
   }
 
+  /**
+   * Use this method to update the leg index of the current {@link DirectionsRoute}
+   * being traveled along.
+   *
+   * An index passed here that is not valid will be ignored.  Please note, the leg index
+   * will automatically increment by default.  To disable this,
+   * use {@link MapboxNavigationOptions#enableAutoIncrementLegIndex()}.
+   *
+   * @param legIndex to be set
+   */
+  public void updateRouteLegIndex(int legIndex) {
```

Comment:
Have we considered returning a `boolean` here instead of "nothing" (`void`)? As a side effect this would allow us to write _State verification_ tests (removing the necessity of Mockito - `verify` / _Behavior_ tests here).